### PR TITLE
B3dm external textures

### LINKED ISSUE #3892: b3dm with external textures reference error
In the 3D Tiles branch, when loading a b3dm file that references external textures like this:

`"images":{"tex_5709034.jpg":{"uri":"textures/tex_5709034.jpg"}`

The request for texture images then looks like this:

`http://localhost:8080/Specs/Data/Cesium3DTiles/Batched/BatchedWithBatchTable/b3dms/R/SW.b3dm/textures/tex_5709034.jpg?v=0.0`

which results in

`Failed to load resource: the server responded with a status of 404 (Not Found)`

The path in not relative to the folder containing the .b3dm file, but to the file itself.

When removing the b3dm header and loading it as an .glb, the textures work fine. Simply adding "../" in front of the path results in not seeing the model at all.

Tested in Chrome and Firefox

### PR DESCRIPTION
**Pull request by sumitshyamsukha**
Fixes #3892

### CODE REVIEW

**Comment by sumitshyamsukha** [comment]
@lilleyse I'm not sure why this PR ended up including all of the previous commits -- I tried cherry-picking, but that didn't work either. Any suggestions on how to submit only the one commit?

**Comment by lilleyse** [comment]
Make sure the base branch that you are merging into is `AnalyticalGraphicsInc:3d-tiles`. You can reopen this PR and change the target branch by clicking the edit button at the top.

**Comment by sumitshyamsukha** [comment]
@lilleyse Thanks so much! I tested this and it seems to fix the issue. Please let me know if there's anything else that needs to be fixed in this PR.

**Comment by lilleyse** [comment]
@sumitshyamsukha Can you change this PR so that it only includes the external textures commit, and open another PR for just the shadows changes.

After that, you will have to resolve the merge conflict. Usually the best approach is to switch to the 3d-tiles branch, `git pull origin 3d-tiles` (make sure that `origin` references the AnaylticalGraphics origin with `git remote -v`), then checkout the `b3dm-external-textures` branch, and do `git merge 3d-tiles`. Edit the code to fix any merge conflicts, add the fixed files, then `git commit`.

**Comment by lilleyse** [comment]
Is this ready? After updating a PR just leave a small comment to let me know to check it out since I usually ignore emails about pushed commits.

**Comment by sumitshyamsukha** [comment]
Sorry for not leaving a comment! I'm still in the process of trying to figure out the merge conflict. I'll leave a comment once it's done!

File: Source/Scene/Instanced3DModel3DTileContent.js [hunk 1]
```diff
@@ -314,7 +316,7 @@ define([
             collectionOptions.url = joinUrls(baseUrl, gltfUrl);
         } else {
             collectionOptions.gltf = gltfView;
-            collectionOptions.basePath = this._url;
+            collectionOptions.basePath = getBaseUri(this._url, false);
```
Comment (lilleyse) [inline Source/Scene/Instanced3DModel3DTileContent.js]: Same comment here.

**Comment by lilleyse** [comment]
I think something went wrong with the merge. The files changed section includes a lot of extraneous files. I would just delete that commit and try the merge again.

This sequence of commands works for me:

```
git checkout 3d-tiles
git pull origin 3d-tiles
git checkout b3dm-external-textures
git merge 3d-tiles
```

**Comment by sumitshyamsukha** [comment]
@lilleyse Updated, should be fine now!

**Comment by lilleyse** [comment]
Thanks @sumitshyamsukha it looks great now!

**Pull request merged**